alda-mode.el

```diff
@@ -54,25 +56,88 @@ If you are experiencing problems, try clearing your history with 'alda-history-c
   "Alda customization options"
   :group 'applications)
 
+(defgroup alda-mode-inf
+  nil
+  "Mode to interact with a Alda interpreter."
+  :group 'Alda
+  :tag "Inferior Alda")
+
 (defcustom alda-binary-location nil
   "Alda binary location for `alda-mode'.
 When set to nil, will attempt to use the binary found on your $PATH.
 This must be a _full_ path to your alda binary."
   :type 'string
   :group 'Alda)
 
-(defcustom alda-ess-keymap t
-  "Whether to use ess keymap in 'alda-mode'.
-When set to nil, will not set any ess keybindings"
-  :type 'boolean
-  :group 'Alda)
+;;;; -- Alda inferior process definitions --
+
+(defconst alda-inf-buffer-name "*inferior-alda*")
+
+(define-derived-mode alda-mode-inf comint-mode "Inferior Alda"
+  "Major mode for interacting with a Alda interpreter.
+
+\\{inferior-alda-mode-map\\}"
+  (define-key alda-mode-inf-map [(meta return)] 'comint-accumulate)
+
+  ;; Comint configuration
+  (make-local-variable 'comint-input-sender)
+  (setq comint-input-sender 'alda-input-sender))
+
+(defun alda-input-sender (proc string)
+  (comint-send-string proc string)
+  (comint-send-string proc "\n"))
+
+;;;###autoload
+(defun alda-interpreter-running-p-1 ()
+  ;; True iff a Alda interpreter is currently running in a buffer.
+  (comint-check-proc alda-inf-buffer-name))
+
+(defun alda-check-or-start-interpreter ()
+  (unless (alda-interpreter-running-p-1)
+    (alda-run-alda (alda-repl))))
 
 (defun alda-location ()
   "Return what 'alda' should be called as in the shell based on 'alda-binary-location' or the path."
   (if alda-binary-location
     alda-binary-location
     (locate-file "alda" exec-path)))
 
+(defun alda-repl ()
+  "Return the 'alda' repl start command"
+  (format "%s repl" (alda-location)))
+
+;;;###autoload
+(defun alda-run-alda (cmd-line)
+  "Run a Alda interpreter in an Emacs buffer"
+  (let* ((alda-interpreter cmd-line)
+         (cmd/args (split-string cmd-line)))
+    (interactive (list (if current-prefix-arg
```

Comment:
I think that moving this interactive inside the let has broken it (it's not callable interactively for me)

Reply:
Sure. just make this function taking no parameter.